### DISCUSSION TITLE
Forward ref to BubbleSelect

### DISCUSSION THREAD

**Discussion by rozig** [radix-ui/primitives#1501] [Help]
Hello,

Is there a way to forwardRef to the BubbleSelect component? I see BubbleSelect component is using `React.forwardRef` but not the `Select` component. Is it intentional?

**Comment by benoitgrelard** [comment]
Hi @rozig,

Yes it is intentional as `BubbleSelect` is an implementation detail and the ref points to what you should be the element on the page (the trigger). In fact, if the component isn't in a `form`, there won't even be a `BubbleSelect`.

**Comment by rozig** [comment]
Hi @benoitgrelard , sorry for the late reply. I'm trying to use select component with [react-hook-form v6](https://react-hook-form.com/v6/api#register) and it was having issues since the ref is only forwarded to the Select component. Is there any way to get it working with it?

**Reply by benoitgrelard** [reply]
You should be able to use their `Controller` API and plug it in. Here's an example with our checkbox: https://github.com/radix-ui/primitives/discussions/874#discussioncomment-1361164

**Reply by rozig** [reply]
Yes, I was able to use `Controller` API and plug it in. I wanted to refactor a component that I'm currently using to @radix-ui/select and I have a plenty of code that is written with `register` API and wanted to check.
